webapp/components/test-runs-query-builder.html

```diff
@@ -383,6 +401,52 @@ <h3>
           labels: (product.labels || []).filter(l => !CHANNELS.has(l)).concat(channel)
         });
       }
+
+      computeVersionsURL(browserName) {
+        const url = new URL('/api/versions', window.location);
+        url.searchParams.set('product', browserName);
+        return url;
+      }
+
+      versionsURLUpdated(url) {
+        fetch(url).then(r => r.json()).then(v => {
+          this.versions = v;
+        });
+      }
+
+      versionsUpdated(version, versions) {
+        if (!versions) {
+          return;
+        }
+        if (version) {
+          versions = versions.filter(s => s.startsWith(version));
+        }
+        versions = versions.slice(0, 10);
+        // Check actually different from current.
+        const curr = this.versionsAutocomplete || [];
+        if (curr.length === versions.length) {
+          let same = true;
```

Comment:
This looks correct. Would the code be simpler if we added versions to a set and check its length?

Reply:
Simplified.